[WIP] Documentation samples for binary classifiers (Static API)

bojanmisic opened the pull request:
[WORK IN PROGRESS]

Adding documentation samples for binary classifiers when using Static API:

1) `SDCA` (using `adult` dataset) - need to use different dataset for testing (instead of `adult.test`) since it is not formatted properly (has trailing `.` at the end of positive lines) and used in ML.NET test scenarios.

sfilipi commented on src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs:
```diff
@@ -17,11 +17,23 @@ public static string DownloadHousingRegressionDataset()
+        /// Downloads the adult train dataset from the ML.NET repo
+        /// </summary>
+        public static string DownloadAdultTrainDataset()
+            => Download("https://raw.githubusercontent.com/dotnet/machinelearning/master/test/data/adult.train", "adult.train");
```
>"https://raw.githubusercontent.com/dotnet/machinelearning/master/test/data/adult.trai [](start = 24, length = 85)

It needs to include the hash, otherwise it might change overtime. 

[https://github.com/dotnet/machinelearning/blob/20c36ca769d6b1f4fc816de4ae5f46266e8edd9b/test/data/adult.test](https://github.com/dotnet/machinelearning/blob/20c36ca769d6b1f4fc816de4ae5f46266e8edd9b/test/data/adult.test)

sfilipi commented on src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs:
```diff
@@ -17,11 +17,23 @@ public static string DownloadHousingRegressionDataset()
+        /// Downloads the adult test dataset from the ML.NET repo
+        /// </summary>
+        public static string DownloadAdultTestDataset()
+            => Download("https://raw.githubusercontent.com/dotnet/machinelearning/master/test/data/adult.test", "adult.test");
```
>https://raw.githubusercontent.com/dotnet/machinelearning/master/test/data/adult.test" [](start = 25, length = 85)

Similar here:
[https://github.com/dotnet/machinelearning/blob/20c36ca769d6b1f4fc816de4ae5f46266e8edd9b/test/data/adult.train](https://github.com/dotnet/machinelearning/blob/20c36ca769d6b1f4fc816de4ae5f46266e8edd9b/test/data/adult.train)

sfilipi commented on docs/samples/Microsoft.ML.Samples/Static/Trainers.cs:
```diff
@@ -191,5 +191,109 @@ public static void LightGbmRegression()
+            // It takes care of logging, exception tracking and as a source of randomness
+            // Using random seed and automatic level of concurrency
+            using (var environment = new ConsoleEnvironment(seed: 0, conc: 0))
+            {
```
this is not necessary, the MLContext below is sufficient.

sfilipi commented on docs/samples/Microsoft.ML.Samples/Static/Trainers.cs:
```diff
@@ -191,5 +191,109 @@ public static void LightGbmRegression()
+                                row.Ethnicity.OneHotEncoding(),
+                                row.Sex.OneHotEncoding(),
+                                row.HoursPerWeek,
+                                row.NativeCountry.OneHotEncoding().SelectFeaturesBasedOnCount(count: 10)),
```
>.SelectFeaturesBasedOnCount(count: 10)) [](start = 66, length = 39)

@shmoradims  to check whether this is necessary.

sfilipi commented:
Thanks for the contribution, @bojanmisic. This looks great, a few small comments. 

If you sync to latest, you'll notice that we are now doing one sample per file, to avoid those pesky line changes in the codebase :)

sharwell commented on docs/samples/Microsoft.ML.Samples/Static/AveragedPerceptron.cs:
```diff
@@ -0,0 +1,115 @@
+
+// NOTE: WHEN ADDING TO THE FILE, ALWAYS APPEND TO THE END OF IT. 
+// If you change the existing content, check that the files referencing it in the XML documentation are still correct, as they reference
+// line by line. 
```
❔  Does it not have the ability to import a named region?

sfilipi replied:
I am actually going to remove the copyright and change  the references to import the whole file. 

---
In reply to: [231153004](https://github.com/dotnet/machinelearning/pull/1456#discussion_r231153004) [](ancestors = 231153004)

bojanmisic commented:
@sfilipi Thank you for the review, I have updated the branch per your suggestions. Also updated the PR description to reflect the things I have changed.

Removed WIP.
